crates/rust-analyzer/src/lsp/capabilities.rs

```diff
@@ -41,7 +41,11 @@ pub fn server_capabilities(config: &Config) -> ServerCapabilities {
         })),
         hover_provider: Some(HoverProviderCapability::Simple(true)),
         completion_provider: Some(CompletionOptions {
-            resolve_provider: Some(config.caps().completions_resolve_provider()),
+            resolve_provider: if config.client_is_helix() || config.client_is_neovim() {
+                None
```

Comment:
This has to be `self.completion_item_edit_resolve().then_some(true)` or similar as it was before https://github.com/rust-lang/rust-analyzer/pull/18589 due to r-a actually always resolving the `additionalTextEdits`.